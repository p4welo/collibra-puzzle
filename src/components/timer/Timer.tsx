import React, { Component } from 'react';
import './Timer.scss';

interface TimerProps {
  time: number
}

export class Timer extends Component<TimerProps> {
  render() {
    return (
        <div className='Timer'>
          <h3>Your time:</h3>
          <h2>{this.props.time} ms</h2>
        </div>
    );
  }
}
