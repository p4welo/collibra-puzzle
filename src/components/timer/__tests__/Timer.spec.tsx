import React from 'react';
import { render } from '@testing-library/react';
import { Timer } from '../Timer';

describe('Timer component', () => {
  it('should display current user time', () => {
    const time = 12345;
    const { queryByText } = render(<Timer time={time}/>);
    expect(queryByText(`12345 ms`)).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const time = 12345;
    const { container } = render(<Timer time={time}/>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
