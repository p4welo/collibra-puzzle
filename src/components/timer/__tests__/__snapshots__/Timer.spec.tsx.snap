// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer component should match snapshot 1`] = `
<div
  class="Timer"
>
  <h3>
    Your time:
  </h3>
  <h2>
    12345
     ms
  </h2>
</div>
`;
